terraform/pipeline/workspace.py

```diff
@@ -0,0 +1,46 @@
+"""Used to set the environment-related variables at runtime (rather than build
+time) so that all the necessary pipeline variables are available."""
+
+import os
+import sys
+
+REASON = os.environ["REASON"]
+# the name of the variable that Azure Pipelines uses for the source branch
+# depends on the type of run, so need to check both
+SOURCE = os.environ.get("OTHER_SOURCE") or os.environ["INDIVIDUAL_SOURCE"]
+TARGET = os.environ["TARGET"]
+
+# the branches that correspond to environments
+ENV_BRANCHES = ["dev", "test", "prod"]
+
+if REASON == "PullRequest" and TARGET in ENV_BRANCHES:
+    # it's a pull request against one of the environment branches, so use the
+    # target branch
+    environment = TARGET
+elif REASON == "IndividualCI" and SOURCE in ENV_BRANCHES:
```

Comment:
Do we want the same `"Manual"` reason here, like from cal-itp/eligibility-server#233

Reply:
Ah yes, thanks, I forgot this script was only introduced by this pull request.